src/sql/workbench/contrib/notebook/test/browser/outputProcessor.test.ts

```diff
@@ -0,0 +1,164 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { nb } from 'azdata';
+import * as op from 'sql/workbench/contrib/notebook/browser/models/outputProcessor';
+import { JSONObject } from 'sql/workbench/services/notebook/common/jsonext';
+import { nbformat as nbformat } from 'sql/workbench/services/notebook/common/nbformat';
+import { getRandomElement } from 'vs/base/common/arrays';
+
+suite('OutputProcessor functions', function (): void {
+	const text = 'An arbitrary text input:!@#$%^&*()_+~`:;,.-_=';
+
+	const arbitraryMetadata = {
+		// arbitrarily construction chart options. It is any JSONObject
+		azdata_chartOptions: {
+			scale: true,
+			dataGrid: false,
+			arbitraryCharProperty: {
+				prop1: 'value1',
+				prop2: {
+					deepProp1: 3
+				}
+			}
+		}
+	};
+
+	suite('getData', function (): void {
+		// data tests
+		for (const outputType of ['execute_result', 'display_data', 'update_display_data'] as nbformat.OutputType[]) {
+			const output = <nbformat.IExecuteResult>{
+				output_type: outputType,
+				data: <op.IMimeBundle>{
+					result: text  // free form fields of the IMimeBundle. This is just an arbitrarily cooked-up hardcoded example for the test.
+				}
+			};
+			test(`test for outputType:${output.output_type}`, () => {
+				verifyGetDataForDataOutput(output);
+			});
+		}
+
+		// error tests
+		for (const traceback of [undefined, [undefined], [], [''], ['tb1:!@#$%^&*()_+~`:;,.-_=', 'tb2:!@#$%^&*()_+~`:;,.-_=']]) {
```

Comment:
let's put a comment here that the string represents arbitrary text

Reply:
did in line in the text of the string.